### PR TITLE
Document feature gates

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,14 +53,14 @@ higher-level APIs:
...
-| [Multicluster Ingress DNS via `external-dns`](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/ingressdns-with-externaldns.md) | Alpha |
-| [Replica Scheduling Preferences](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#replicaschedulingpreference) | Alpha |
+| Feature | Maturity | Feature Gate | Default |
+|---------|----------|--------------|---------|
+| [Push propagation of arbitrary types to remote clusters](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#example) | Alpha | PushReconciler | true |
+| [CLI utility (`kubefedctl`)](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#operations) | Alpha | | |
+| [Generate Federation APIs without writing code](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#enabling-federation-of-an-api-type) | Alpha | | |
+| [Multicluster Service DNS via `external-dns`](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/servicedns-with-externaldns.md) | Alpha | CrossClusterServiceDiscovery | true |
```
Comment (marun) [inline README.md]: @shashidharatd Should MC ingress and service dns continue to be enabled by default post-beta?

File: README.md [hunk 2]
```diff
@@ -53,14 +53,14 @@ higher-level APIs:
...
+| [Multicluster Ingress DNS via `external-dns`](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/ingressdns-with-externaldns.md) | Alpha | FederatedIngress | true |
+| [Replica Scheduling Preferences](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#replicaschedulingpreference) | Alpha | SchedulerPreferences | true |
```
Comment (marun) [inline README.md]: @irfanurrehman Should scheduling continue to be enabled by default post-beta?

**Comment by marun** [comment]
/approve
/lgtm

Thank you!

File: README.md (hunk 2 above)
Comment (irfanurrehman) [reply README.md]: I think it would be ok to keep is disabled by default.

**Pull request merged**